### ISSUE TITLE
null vs None - more Pythonic bindings?

### ISSUE THREAD

**Issue by prokopst** [touilleMan/godot-python#82]
When I used by mistake wrong path in `Node.get_node`, for example:
```
self.get_node('/root/quite/wrong/path')
```
An instance of object `<godot.hazmat.base.Object object at 0x000001382FAE2B38>` was returned instead of `None`. Apparently the object returned by `Node.get_node` is `null`, as documentation suggests:

> Fetches a node. The NodePath can be either a relative path (from the current node) or an absolute path (in the scene tree) to a node. **If the path does not exist, a `null instance` is returned** and attempts to access it will result in an “Attempt to call <method> on a null instance.” error.

http://docs.godotengine.org/en/3.0/classes/class_node.html#class-node-get-node

The object does not work with `if xzy:` and I haven't found any null within `godot.bindings`, but I would like to raise a different topic:

I wonder if the project aims to do Python bindings that strictly follow what Godot does, or to be more Pythonic.

**Comment by touilleMan** [comment]
You're right, this `null` object should be converted to a nice Python `None`.
Actually there is already code dealing with this (see https://github.com/touilleMan/godot-python/blob/e11fde2bc7e810e3194bc6ea6035ddd2dc87aa16/pythonscript/embedded/godot/hazmat/tools.py#L366). So this code should be corrected to also take into account this `null` passed as object.

> I wonder if the project aims to do Python bindings that strictly follow what Godot does, or to be more Pythonic.

The idea is to provide a Pythonic API (well while still trying to stay close to the Godot API) by default, while also providing the `godot.hazmat` low-level module which give you raw access to the GDNative API.

**Comment by prokopst** [comment]
Great, thanks for the clarification.

> So this code should be corrected to also take into account this `null` passed as object.

I wanted to fix it, but I just can't figure out what's going on. `Node::get_node` should return `NULL` when nothing is found:
https://github.com/godotengine/godot/blob/e9e5aa617e7f90bd1706f38109f4ebf5f62ab58a/scene/main/node.cpp#L1364
https://github.com/godotengine/godot/blob/612ab4bbc6f2396f4dcd68c3f142f7dfa2f5f0a5/core/error_macros.h#L224

But it just doesn't.

**Comment by garyo** [comment]
@prokopst I don't know how to fix it either, but that code you're looking at is for when there is actually no node found. I think in this case there is a valid object; it's just the "null object". And in any case it has to work not just for Nodes but any Object. (The null object is not a Node; it's of class Object apparently.)
